src/browser/session.zig

```diff
@@ -116,12 +126,12 @@ pub const Session = struct {
         // it isn't null!
         std.debug.assert(self.page != null);
 
-        // can't use the page arena, because we're about to reset it
-        // and don't want to use the session's arena, because that'll start to
-        // look like a leak if we navigate from page to page a lot.
-        var buf: [2048]u8 = undefined;
-        var fba = std.heap.FixedBufferAllocator.init(&buf);
-        const url = try self.page.?.url.resolve(fba.allocator(), url_string);
+        _ = self.browser.transfer_arena.reset(.{ .retain_with_limit = 1 * 1024 * 1024 });
```

Comment:
It surprises me the `transfer_arena` is reset at the beginning of the page navigate.
What about instead reset the transfer_arena after the end of `pageNavigate`? The data must live longer?

If we reset it at the beginning of the `pageNavigate` it must be explicit we shouldn't use it outside the function...

Reply:
I wasn't thinking. I originally had the reset _inside_ `page.navigate`, but I was happier with the `page` knowing nothing about it. Don't know why I didn't think of just defer inside the function.